pkg/csi/service/wcp/controller.go

```diff
@@ -800,6 +800,17 @@ func (c *controller) ControllerPublishVolume(ctx context.Context, req *csi.Contr
 
 		vmuuid, err := getVMUUIDFromK8sCloudOperatorService(ctx, req.VolumeId, req.NodeId)
 		if err != nil {
+			if e, ok := status.FromError(err); ok {
+				switch e.Code() {
+				case codes.NotFound:
+					return nil, csifault.CSIVmUuidNotFoundFault, logger.LogNewErrorCodef(log, codes.Internal,
```

Comment:
We're just relying on status code and not checking the error type as you proposed in your design.
We can throw `NotFound` from multiple places, say also when pod is not found. How do you plan to differentiate?

Reply:
k8scloudoperator running in syncer internally will use `ErrVMUuidNotFound` to identify vmuuid not found error. However we can't return this custom error in grpc error codes from syncer, because grpc error codes must be defined in thie table : https://grpc.github.io/grpc/core/md_doc_statuscodes.html.  So we will return `NotFound` error code in the grpc error response when we encounter `ErrVMUuidNotFound`. The `NotFound` grpc error code will then be checked by controller AttachVolume to identify vmuuid not found issues and returns the non-storage fault.

I have added comments in `getVMUUIDFromK8sCloudOperatorService` to make sure `NotFound` will be returned only when VMUUID is not found. For other errors like pod not found, we should return Internal.

Reply:
Makes sense. Thanks.